File: internal/models/notification/wechatwork.go
// Package notification 定义消息通知相关的数据模型
// 本文件包含企业微信相关的模型定义
package notification

import (
	"time"
)

// ==================== 企业微信模型 ====================

// WechatWorkApp 企业微信应用模型
// 存储企业微信应用配置
type WechatWorkApp struct {
	ID          uint      `gorm:"primarykey" json:"id"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
	Name        string    `gorm:"size:100;not null" json:"name"`                     // 应用名称
	CorpID      string    `gorm:"column:corp_id;size:100;not null" json:"corp_id"`   // 企业ID
	AgentID     int64     `gorm:"column:agent_id;not null" json:"agent_id"`          // 应用 Agent ID
	Secret      string    `gorm:"column:secret;size:200;not null" json:"secret"`     // 应用密钥
	Project     string    `gorm:"size:100" json:"project"`                           // 关联项目
	Status      string    `gorm:"size:20;default:'active';not null" json:"status"`   // 状态
	Description string    `gorm:"type:text" json:"description"`                      // 描述
	IsDefault   bool      `gorm:"column:is_default;default:false" json:"is_default"` // 是否默认应用
	CreatedBy   *uint     `gorm:"column:created_by" json:"created_by"`
}

// TableName 指定表名
func (WechatWorkApp) TableName() string {
	return "wechat_work_apps"
}

// WechatWorkBot 企业微信机器人模型
// 存储企业微信 Webhook 机器人配置
type WechatWorkBot struct {
	ID          uint      `gorm:"primarykey" json:"id"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
	Name        string    `gorm:"size:100;not null" json:"name"`                   // 机器人名称
	WebhookURL  string    `gorm:"size:500;not null" json:"webhook_url"`            // Webhook URL
	Description string    `gorm:"type:text" json:"description"`                    // 描述
	Status      string    `gorm:"size:20;default:'active';not null" json:"status"` // 状态
	CreatedBy   *uint     `gorm:"index" json:"created_by"`
}

// TableName 指定表名
func (WechatWorkBot) TableName() string {
	return "wechat_work_bots"
}

// WechatWorkMessageLog 企业微信消息发送记录
// 记录所有通过企业微信发送的消息
type WechatWorkMessageLog struct {
	ID        uint      `gorm:"primarykey" json:"id"`
	CreatedAt time.Time `json:"created_at"`
	MsgType   string    `gorm:"size:50;not null" json:"msg_type"`        // 消息类型
	ToUser    string    `gorm:"size:500" json:"to_user"`                 // 接收用户
	ToParty   string    `gorm:"size:500" json:"to_party"`                // 接收部门
	ToTag     string    `gorm:"size:500" json:"to_tag"`                  // 接收标签
	Content   string    `gorm:"type:text" json:"content"`                // 消息内容
	Title     string    `gorm:"size:200" json:"title"`                   // 标题
	Source    string    `gorm:"size:50" json:"source"`                   // 来源
	Status    string    `gorm:"size:20;default:'success'" json:"status"` // 状态
	ErrorMsg  string    `gorm:"type:text" json:"error_msg"`              // 错误信息
	AppID     uint      `gorm:"column:app_id" json:"app_id"`             // 使用的企业微信应用ID
}

// TableName 指定表名
func (WechatWorkMessageLog) TableName() string {
	return "wechat_work_message_logs"
}

File: web/src/services/wechatwork.ts
import request from './api'
import type { ApiResponse } from '../types'

// 企业微信应用
export interface WechatWorkApp {
  id?: number
  name: string
  corp_id: string
  agent_id: number
  secret: string
  project: string
  description?: string
  status: string
  is_default: boolean
  created_at?: string
  updated_at?: string
}

// 企业微信机器人
export interface WechatWorkBot {
  id?: number
  name: string
  webhook_url: string
  description?: string
  status: string
  created_at?: string
  updated_at?: string
}

// 企业微信消息日志
export interface WechatWorkMessageLog {
  id: number
  created_at: string
  msg_type: string
  target: string
  content: string
  title?: string
  source: string
  status: string
  error_msg?: string
  app_id?: number
}

// 企业微信用户
export interface WechatWorkUser {
  userid: string
  name: string
  mobile?: string
  email?: string
  avatar?: string
  department?: number[]
}

// 发送应用消息请求
export interface SendMessageRequest {
  app_id?: number
  to_user: string
  to_party?: string
  to_tag?: string
  msg_type: 'text' | 'markdown' | 'textcard' | 'news'
  content: string
  title?: string
  url?: string
}

// 发送Webhook请求
export interface SendWebhookRequest {
  bot_id: number
  msg_type: 'text' | 'markdown' | 'news'
  content: { content?: string }
  mentioned_list?: string[]
  mentioned_mobile_list?: string[]
}


// 应用绑定信息
export interface AppBindings {
  jenkins_instances: Array<{ id: number; name: string; url: string }>
  k8s_clusters: Array<{ id: number; name: string }>
}

// 企业微信应用管理 API
export const wechatworkAppApi = {
  list: (page = 1, pageSize = 100): Promise<ApiResponse<{ list: WechatWorkApp[]; total: number }>> => {
    return request.get('/wechatwork/app', { params: { page, page_size: pageSize } })
  },

  get: (id: number): Promise<ApiResponse<WechatWorkApp>> => {
    return request.get(`/wechatwork/app/${id}`)
  },

  create: (data: Partial<WechatWorkApp>): Promise<ApiResponse<WechatWorkApp>> => {
    return request.post('/wechatwork/app', data)
  },

  update: (id: number, data: Partial<WechatWorkApp>): Promise<ApiResponse<WechatWorkApp>> => {
    return request.put(`/wechatwork/app/${id}`, data)
  },

  delete: (id: number): Promise<ApiResponse> => {
    return request.delete(`/wechatwork/app/${id}`)
  },

  setDefault: (id: number): Promise<ApiResponse> => {
    return request.post(`/wechatwork/app/${id}/default`)
  },

  getBindings: (id: number): Promise<ApiResponse<AppBindings>> => {
    return request.get(`/wechatwork/app/${id}/bindings`)
  }
}

// 企业微信机器人管理 API
export const wechatworkBotApi = {
  list: (page = 1, pageSize = 100): Promise<ApiResponse<{ list: WechatWorkBot[]; total: number }>> => {
    return request.get('/wechatwork/bot', { params: { page, page_size: pageSize } })
  },

  get: (id: number): Promise<ApiResponse<WechatWorkBot>> => {
    return request.get(`/wechatwork/bot/${id}`)
  },

  create: (data: Partial<WechatWorkBot>): Promise<ApiResponse<WechatWorkBot>> => {
    return request.post('/wechatwork/bot', data)
  },

  update: (id: number, data: Partial<WechatWorkBot>): Promise<ApiResponse<WechatWorkBot>> => {
    return request.put(`/wechatwork/bot/${id}`, data)
  },

  delete: (id: number): Promise<ApiResponse> => {
    return request.delete(`/wechatwork/bot/${id}`)
  }
}

// 企业微信消息 API
export const wechatworkApi = {
  sendMessage: (data: SendMessageRequest): Promise<ApiResponse> => {
    return request.post('/wechatwork/send-message', data)
  },

  sendWebhook: (data: SendWebhookRequest): Promise<ApiResponse> => {
    return request.post('/wechatwork/send-webhook', data)
  },

  searchUser: (query: string, appId?: number): Promise<ApiResponse<WechatWorkUser[]>> => {
    return request.post('/wechatwork/user/search', { query, app_id: appId })
  }
}

// 企业微信消息日志 API
export const wechatworkLogApi = {
  list: (page = 1, pageSize = 20, msgType = '', source = ''): Promise<ApiResponse<{ list: WechatWorkMessageLog[]; total: number }>> => {
    return request.get('/wechatwork/logs', { params: { page, page_size: pageSize, msg_type: msgType, source } })
  }
}
